core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -436,6 +486,29 @@ private void recursiveDelete(Path path) {
         }
     }
 
+    /**
+     * Check if the segment directory has any outstanding ".pending" marker files in a ".pending" directory.
+     */
+    private boolean segmentHasPendingTasks(int walId, int segmentId) {
+        final Path pendingPath = setSegmentPendingPath(tableToken, walId, segmentId);
+        final long p = ff.findFirst(pendingPath);
+        if (p > 0) {
```

Comment:
Does `p<=0` mean directory does not exit? Is it covered by tests?

Reply:
`findFirst` returns `p == 0` if the directory does not exist (i.e. `errno == ENOENT`).
The case of the missing `.pending` directory is already covered by all other pre-existing test cases.